AsyncDisplayKit/Details/ASRangeController.mm

```diff
@@ -145,6 +149,10 @@ - (void)updateCurrentRangeWithMode:(ASLayoutRangeMode)rangeMode
     _didUpdateCurrentRange = YES;
 
     [self setNeedsUpdate];
+  } else if (_needsRangeUpdate) {
+    // If _needsRangeUpdate is YES, but we don't set didUpdateCurrentRange, it
+    // will be overridden.
+    _didUpdateCurrentRange = YES;
```

Comment:
I suppose so, yes.  We should also rename the instance variable to something more descriptive like, _preserveRangeMode or _rangeModeSetManually